file/provider/expand_volume.go

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2021 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package provider ...
+package provider
+
+import (
+	"time"
+
+	userError "github.com/IBM/ibmcloud-volume-file-vpc/common/messages"
+	"github.com/IBM/ibmcloud-volume-file-vpc/common/vpcclient/models"
+	"github.com/IBM/ibmcloud-volume-interface/lib/metrics"
+	"github.com/IBM/ibmcloud-volume-interface/lib/provider"
+	"go.uber.org/zap"
+)
+
+// GiB ...
+const (
+	GiB = 1024 * 1024 * 1024
+)
+
+func (vpcs *VPCSession) ExpandVolume(expandVolumeRequest provider.ExpandVolumeRequest) (size int64, err error) {
+	vpcs.Logger.Debug("Entry of ExpandVolume method...")
+	defer vpcs.Logger.Debug("Exit from ExpandVolume method...")
+	defer metrics.UpdateDurationFromStart(vpcs.Logger, "ExpandVolume", time.Now())
+
+	// Get volume details
+	existingVolume, err := vpcs.GetVolume(expandVolumeRequest.VolumeID)
+	if err != nil {
+		return -1, err
+	}
+	// Return existing Capacity if its greater or equal to expandable size
+	if existingVolume.Capacity != nil && int64(*existingVolume.Capacity) >= expandVolumeRequest.Capacity {
```

Comment:
how the use will be aware that expected size if not expanded as it less than current size? is this same behaviour from vpc?

Reply:
When we try to edit the pvc with newSize less than the current size, then k8s itself gives an error 

```
# persistentvolumeclaims "file-exp1" was not valid:
# * spec.resources.requests.storage: Forbidden: field can not be less than previous value
#
```

Since VPC doesnot support downsizing the vol size, we are putting a precautionary check if the newSize is less or equal to currentSize then simply return the currentSize

Reply:
@prankulmahajan did you check if VPC is returning any error code if we try downsizing that we can return to the user as backend error instead of handling. If in future we need to support this we wont require any code changes.

Reply:
Tried curl command to try downsizing.
```
{
    "errors": [
        {
            "code": "shares_capacity_invalid",
            "message": "The new share size '10' must be greater than the current share size '250'.",
            "more_info": "Change the 'size' field in the request to be greater than current share size"
        }
    ],
    "trace": "b5697643-adb4-485c-b62c-25d8fda7c75c"
}
```

Reply:
I can remove this condition and let VPC handle this

Reply:
Just confirm with @arahamad once if he also agrees and also make sure in driver we get this error in the backenderror  in the logs and in the describe of pod/pvc

Reply:
Sure

Reply:
Add warn logger